From 7ec9e7db89c00266d9ae6b1389ec4e839ae007f4 Mon Sep 17 00:00:00 2001
From: Suman Anna <s-anna@ti.com>
Date: Wed, 10 May 2017 14:19:15 -0500
Subject: [PATCH 13/16] Revert "TEMP: remoteproc/pru: enable ethernet fw for
 PRUSS1 on AM571x IDK"

This reverts commit 2f7dc44e8e030bb9d18cd6bac2892fa41c34ea35.

The PRU Ethernet driver will use the rproc_set_firmware() API to
configure the appropriate firmware names to be used by the remoteproc
core for loading and booting the PRU remoteprocs. Undo the changes
added previously in PRU remoteproc driver in commit 2f7dc44e8e03
("TEMP: remoteproc/pru: enable ethernet fw for PRUSS1 on AM571x IDK")
to perform the same. The AM571x IDK vs AM572x IDK differences will
be inherent due to the presence/absence of the corresponding prueth
DT nodes.

NOTE:
The internal pinmux programming required for AM571x IDK is still in
place in the PRU remoteproc driver.

Signed-off-by: Suman Anna <s-anna@ti.com>
Signed-off-by: Jacob Stiffler <j-stiffler@ti.com>
---
 drivers/remoteproc/pru_rproc.c | 11 +++--------
 1 file changed, 3 insertions(+), 8 deletions(-)

diff --git a/drivers/remoteproc/pru_rproc.c b/drivers/remoteproc/pru_rproc.c
index 3998160..c31a5b6 100644
--- a/drivers/remoteproc/pru_rproc.c
+++ b/drivers/remoteproc/pru_rproc.c
@@ -73,7 +73,7 @@ enum pru_mem {
  */
 struct pru_private_data {
 	u32 id;
-	int caps;
+	const int caps;
 	const char *fw_name;
 	const char *eth_fw_name;
 };
@@ -680,7 +680,6 @@ static const struct pru_private_data *pru_rproc_get_private_data(
 {
 	const struct pru_match_private_data *data;
 	const struct of_device_id *match;
-	struct pru_private_data *pdata = NULL;
 
 	match = of_match_device(pru_rproc_match, &pdev->dev);
 	if (!match)
@@ -688,14 +687,10 @@ static const struct pru_private_data *pru_rproc_get_private_data(
 
 	for (data = match->data; data && data->device_name; data++) {
 		if (!strcmp(dev_name(&pdev->dev), data->device_name))
-			pdata = data->priv_data;
+			return data->priv_data;
 	}
 
-	/* fixup PRU capability differences between AM571x and AM572x IDKs */
-	if (pdata && of_machine_is_compatible("ti,am5718-idk"))
-		pdata->caps = PRU_FUNC_CAPS_ETHERNET;
-
-	return pdata;
+	return NULL;
 }
 
 static int pru_rproc_probe(struct platform_device *pdev)
-- 
1.9.1

